core/00_primordials.js

```diff
@@ -31,508 +31,506 @@
-
-  primordials.PromisePrototypeCatch = (thisPromise, onRejected) =>
-    PromisePrototypeThen(thisPromise, undefined, onRejected);
-
-  /**
-   * Creates a Promise that is resolved with an array of results when all of the
-   * provided Promises resolve, or rejected when any Promise is rejected.
-   * @param {unknown[]} values An array of Promises.
-   * @returns A new Promise.
-   */
-  primordials.SafePromiseAll = (values) =>
-    // Wrapping on a new Promise is necessary to not expose the SafePromise
-    // prototype to user-land.
-    new Promise((a, b) =>
-      SafePromise.all(
-        ArrayPrototypeMap(
-          values,
-          (p) => {
-            if (ObjectPrototypeIsPrototypeOf(PromisePrototype, p)) {
-              return new SafePromise((c, d) => PromisePrototypeThen(p, c, d));
-            }
-            return p;
-          },
-        ),
-      ).then(a, b)
-    );
-
-  /**
-   * Attaches a callback that is invoked when the Promise is settled (fulfilled or
-   * rejected). The resolved value cannot be modified from the callback.
-   * Prefer using async functions when possible.
-   * @param {Promise<any>} thisPromise
-   * @param {() => void) | undefined | null} onFinally The callback to execute
-   *        when the Promise is settled (fulfilled or rejected).
-   * @returns A Promise for the completion of the callback.
-   */
-  primordials.SafePromisePrototypeFinally = (thisPromise, onFinally) =>
-    // Wrapping on a new Promise is necessary to not expose the SafePromise
-    // prototype to user-land.
-    new Promise((a, b) =>
-      new SafePromise((a, b) => PromisePrototypeThen(thisPromise, a, b))
-        .finally(onFinally)
-        .then(a, b)
-    );
-
-  // Create getter and setter for `queueMicrotask`, it hasn't been bound yet.
-  let queueMicrotask = undefined;
-  ObjectDefineProperty(primordials, "queueMicrotask", {
-    get() {
-      return queueMicrotask;
-    },
-  });
-  primordials.setQueueMicrotask = (value) => {
-    if (queueMicrotask !== undefined) {
-      throw new Error("queueMicrotask is already defined");
-    }
-    queueMicrotask = value;
-  };
+// Renaming from `eval` is necessary because otherwise it would perform direct
+// evaluation, allowing user-land access to local variables.
+// This is because the identifier `eval` is somewhat treated as a keyword
+primordials.indirectEval = eval;
 
-  // Renaming from `eval` is necessary because otherwise it would perform direct
-  // evaluation, allowing user-land access to local variables.
-  // This is because the identifier `eval` is somewhat treated as a keyword
-  primordials.indirectEval = eval;
+ObjectSetPrototypeOf(primordials, null);
+ObjectFreeze(primordials);
 
-  ObjectSetPrototypeOf(primordials, null);
-  ObjectFreeze(primordials);
+globalThis.__bootstrap = { primordials };
 
-  // Provide bootstrap namespace
-  globalThis.__bootstrap = { primordials };
-})();
+// Provide bootstrap namespace
+export { primordials };
```

Comment:
Doesn't matter much but would it be cuter to export the primordials as named exports, instead of this named object export?

Reply:
Same could be done for at least some common parts of core, like ops.

Reply:
definitively; just going to be a cleanup later or a follow-up

Reply:
Sounds reasonable :)

Reply:
primordials cannot be exported as named exports due to them being generated dynamically.

Reply:
They could but it would be painful